### DISCUSSION TITLE
Shiro Security for VS code karavan

### DISCUSSION THREAD

**Discussion by samar-elsayed** [apache/camel-karavan#1183] [Q&A] [answered]
Hi,

is there something / component like Shiro Security available to use in Karavan VScode? 
I am using the latest version

**Comment by davsclaus** [comment]
Can you explain more what you want to do with this in VSCode? Where in Karavan do you see it can be used?

There is camel-shiro
https://camel.apache.org/components/4.4.x/others/shiro.html

Which you can use as a route policy. You can configure route policy in karavan.

**Comment by samar-elsayed** [comment]
@davsclaus  I would like to create an API that users can use to generate tokens using **Shiro** and then these tokens can be used to secure the calls done to other APIs (which include Routes) and all of that will be done by Karavan VS code

**so could you please show me an example of how to do that route policy inside Karavan VSCode ?** 

- for example what is the word **zone1** inside the example below mentioned in the documentation :
does **Zone** means **Route ID / name** ?

```
[users]
# user 'ringo' with password 'starr' and the 'sec-level1' role
ringo = starr, sec-level1
george = harrison, sec-level2
john = lennon, sec-level3
paul = mccartney, sec-level3

[roles]
# 'sec-level3' role has all permissions, indicated by the
# wildcard '*'
sec-level3 = *

# The 'sec-level2' role can do anything with access of permission
# readonly (*) to help
sec-level2 = zone1:*

# The 'sec-level1' role can do anything with access of permission
# readonly
sec-level1 = zone1:readonly:*
```

- is there a certain path where I need to put that INI file inside the integration project ?

Thanks and I am waiting for your response

**Reply by davsclaus** [reply]
The Camel Shiro component is only for using Shiro to secure Camel routes. Not for security in general.
And its not in use much by users, as you don't need this feature usually.

You secure your APIs using a platform or API gateway/management and other means.

So what you want sounds a bit like you need to write some custom code for doing your use-case. Karavan can do the Camel part if you really need Camel, but what do you need it for?

**Reply by samar-elsayed** [reply, marked as answer]
My team and I are working on using Karavan to create integration projects/services that other components will need to authenticate before using them so we were thinking about also using Karavan to create a central db-less service to be used for Authentication and Authorization 

we have Kong already and it seems that we can have what we need there except that it will require a db but it is doable so we will explore that option
